source/common/upstream/outlier_detection_impl.cc

```diff
@@ -471,8 +475,28 @@ void DetectorImpl::ejectHost(HostSharedPtr host,
           runtime_.snapshot().getInteger(BaseEjectionTimeMsRuntime, config_.baseEjectionTimeMs()));
       const std::chrono::milliseconds max_eject_time = std::chrono::milliseconds(
           runtime_.snapshot().getInteger(MaxEjectionTimeMsRuntime, config_.maxEjectionTimeMs()));
-      if ((host_monitors_[host]->ejectTimeBackoff() * base_eject_time) <
-          (max_eject_time + base_eject_time)) {
+      const std::chrono::milliseconds max_eject_time_jitter =
+          std::chrono::milliseconds(runtime_.snapshot().getInteger(
+              MaxEjectionTimeJitterMsRuntime, config_.maxEjectionTimeJitterMs()));
+
+      // Generate random jitter so that not all hosts uneject at the same time,
+      // which could possibly generate a connection storm
+
+      // std::chrono::milliseconds receives a signed integer as input, so we ensure that our random
+      // random value is in the range [0, std::numeric_limits<long>::max())
+      long random_long =
+          static_cast<long>(random_generator_() % (std::numeric_limits<long>::max()));
+      // cap the jitter at max_eject_time_jitter
+      // if max_eject_time_jitter is 0, the jitter will also equal 0 because we apply the
+      // modulo operator with 1.
+      const std::chrono::milliseconds jitter =
+          std::chrono::milliseconds(random_long) %
+          (max_eject_time_jitter + std::chrono::milliseconds(1));
+      // save the jitter on the current host_monitor
+      host_monitors_[host]->setJitter(jitter);
+
+      if ((host_monitors_[host]->ejectTimeBackoff() * base_eject_time + jitter) <
+          (max_eject_time + base_eject_time + jitter)) {
```

Comment:
why is `jitter` aded to both sides of the comparison?

Reply:
This is so that that the inequality continues to behave exactly as it did without the jitter.
Otherwise, on some hosts the branch will be taken and on others not depending on a random value.
I think of the jitter as a small random value that will be added to the ejection time but will not effect any other behavior.

Reply:
What I'm missing here is why `if (A + C < B + C)` is better than `if (A < B)`?

Reply:
Fixed

Reply:
@adisuissa Can I please get an invite to the envoy dev slack channel to asheryer@amazon.com?